File: LeetCodes/reverse-words-in-a-string-iii/10-1-2023, 11_19_35 AM/reverse-words-in-a-string-iii.js
// Link: https://leetcode.com/problems/reverse-words-in-a-string-iii

/** Submission Info:
 *
 *  AUTHOR: the_247 | Raj Patel
 *  TITLE: reverse-words-in-a-string-iii
 *  RUNTIME: 67 ms
 *  MEMORY: 48 MB
 *  DATE: 10-1-2023, 11:19:35 AM
 *
 */

// Solution:

var reverseWords = function(s) {
    const words = s.split(' '); 
    for (let i = 0; i < words.length; i++) {
        words[i] = words[i].split('').reverse().join('');
    }
    return words.join(' ');
};

File: LeetCodes/reverse-words-in-a-string-iii/9-16-2023, 2_53_37 PM/reverse-words-in-a-string-iii.cpp
// Link: https://leetcode.com/problems/reverse-words-in-a-string-iii

/** Submission Info:
 *
 *  AUTHOR: the_247 | Raj Patel
 *  TITLE: reverse-words-in-a-string-iii
 *  RUNTIME: 18 ms
 *  MEMORY: 11.7 MB
 *  DATE: 9-16-2023, 2:53:37 PM
 *
 */

// Solution:

class Solution {
public:
    string reverseWords(string s) {
        string str = "", ans = "";
        for(auto &ch: s) {
            if(ch == ' ') {
                reverse(str.begin(), str.end());
                ans += str;
                ans += " ";
                str = "";
            } else {
                str += ch;
            }
        }
        if(str.length() > 0) {
            reverse(str.begin(), str.end());
            ans += str;
        }
        return ans;
    }
};
